ext/js/dom/document-util.js

```diff
@@ -113,27 +113,42 @@ export class DocumentUtil {
         const text = source.text();
         const textLength = text.length;
         const textEndAnchor = textLength - endLength;
-        let pos1 = startLength;
-        let pos2 = textEndAnchor;
+
+        /** Relative start position of the sentence (inclusive). */
+        let cursorStart = startLength;
+        /** Relative end position of the sentence (exclusive). */
+        let cursorEnd = textEndAnchor;
 
         // Move backward
         let quoteStack = [];
-        for (; pos1 > 0; --pos1) {
-            const c = text[pos1 - 1];
+        for (; cursorStart > 0; --cursorStart) {
+            let c = text[cursorStart - 1];
             if (c === '\n' && terminateAtNewlines) { break; }
 
             if (quoteStack.length === 0) {
-                const terminatorInfo = terminatorMap.get(c);
+                let terminatorInfo = terminatorMap.get(c);
                 if (typeof terminatorInfo !== 'undefined') {
-                    if (terminatorInfo[0]) { --pos1; }
+                    // Include the previous character while it is a terminator character and is included at start.
+                    while (typeof terminatorInfo !== 'undefined' && terminatorInfo[0] && cursorStart > 0) {
```

Comment:
I wasn't going to comment on this since I thought typescript wouldn't behave well with this change, but you can do a minor optimization of these while loops by removing the undefined check on the first loop, since that's guaranteed by the enclosing `if` statement.

```js
while (terminatorInfo[0] && cursorStart > 0) { // <-- Removed here
    --cursorStart;
    if (cursorStart === 0) { break; }
    c = text[cursorStart - 1];
    terminatorInfo = terminatorMap.get(c);
    if (typeof terminatorInfo === 'undefined') { break; } // <-- Moved here
}
```